src/include/ipc/dai-intel.h

```diff
@@ -196,4 +197,27 @@ struct sof_ipc_dai_dmic_params {
 	struct sof_ipc_dai_dmic_pdm_ctrl pdm[0];
 } __attribute__((packed));
 
+struct sof_ipc_dai_multidai_ext {
+	uint32_t type;
+	uint32_t dai_index;
+
+	uint32_t reserved[6];
+	union {
+		struct sof_ipc_dai_ssp_params ssp;
+		struct sof_ipc_dai_dmic_params dmic;
+		struct sof_ipc_dai_hda_params hda;
+		struct sof_ipc_dai_alh_params alh;
+	};
+} __attribute__((packed));
+
+struct sof_ipc_dai_multidai_params {
+	struct sof_ipc_hdr hdr;
+	uint32_t num_dais;
+
+	uint32_t reserved[6];
+
+	struct sof_ipc_stream_map stream_map;
+	struct sof_ipc_dai_multidai_ext ext[0];
+} __attribute__((packed));
+
```

Comment:
Why would we not want to make this generic in dai.h ? 
I would prefer we have a parent params structure that would contain an array of sof_ipc_dai_config. This would allow reuse a lot of existing logic in the driver/fw and also mean we could pack in different types together.

Reply:
I made it multidai specific to avoid major ABI changes.

Reply:
Should juyst be a MINOR ABI change. We need to create a new struct that can contain sof_ipc_dai_config

Reply:
Oh, sorry, I misunderstood what you mean. Now that you mentioned it, yes, I agree.

Reply:
Having thought about this, I don't thing there's a sane way to do that. Such a mechanic introduces circular dependency between sof_ipc_dai_multidai_params and sof_ipc_dai_config.

Reply:
How will we have a circular dependency ? They should both be defined in the same header ?

Reply:
They both would have an instance (or more) of each other:
sof_ipc_dai_multidai_params having a flex array of sof_ipc_dai_configs and sof_ipc_dai_config having sof_ipc_dai_multidai_params in the union

Reply:
oh, I would not expect sof_ipc_dai_config to contain the multidai type.
I would rename sof_ipc_dai_multidai_params to sof_ipc_dai_multidai_config which would be a new toplevel IPC and can contain an array of  sof_ipc_dai_config

Reply:
Ha, thinking out of the box. Didn't even cross my mind. That is a possibility indeed.
@plbossart @bardliao what do you think?

Reply:
This is MINOR ABI update and will require a mechanical kernel update too (i.e. if we detect > 1 DAI config then use new IPC)

Reply:
Changes applied.